clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -1984,6 +1984,15 @@ llvm::Constant *ConstantEmitter::emitForMemory(CodeGenModule &CGM,
     return Res;
   }
 
+  // In HLSL bool vectors are stored in memory as a vector of i32
+  if (destType->isExtVectorBoolType() && CGM.getContext().getLangOpts().HLSL) {
```

Comment:
```suggestion
  if (destType->isExtVectorBoolType() && !destType->isPackedVectorBoolType(Ctx)) {
```

Same for other places where you explicitly check for HLSL.